### PR TITLE
minimally functional resource_user implementation

### PR DESCRIPTION
**Pull request by sfc-gh-gbutzi**

### CODE REVIEW

File: snowflake/provider.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ func Provider() terraform.ResourceProvider {
 			"snowflake_table":    resourceSnowflakeTable(),
 			"snowflake_pipe":     resourceSnowflakePipe(),
 			"snowflake_view":     resourceSnowflakeView(),
+            "snowflake_user":     resourceSnowflakeUser(),
```
Comment (Preston4tw) [inline snowflake/provider.go]: This looks out of indent alignment, `go fmt` should fix this

File: snowflake/resource_snowflake_user.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		return err
+	}
+	d.SetId(name)
+	return nil
+}
+
+func resourceSnowflakeUserRead(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := d.Id()
+	databaseInfo, err := showUser(db, name)
```
Comment (Preston4tw) [inline snowflake/resource_snowflake_user.go]: databaseInfo should presumably be userInfo

File: snowflake/resource_snowflake_user.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	if err != nil {
+		return err
+	}
+	d.Set("name", databaseInfo.name)
+	return nil
+}
+
+func resourceSnowflakeUserUpdate(d *schema.ResourceData, meta interface{}) error {
```
Comment (Preston4tw) [inline snowflake/resource_snowflake_user.go]: Lets not even define an update function.

File: snowflake/row_structs.go [hunk 4]
```diff
@@ -5,6 +5,34 @@ import (
 	"time"
 )
 
+type showUserRow struct {
```
Comment (Preston4tw) [inline snowflake/row_structs.go:8]: more `go fmt`

**Review by Preston4tw — CHANGES_REQUESTED** [review]

**Comment by sfc-gh-gbutzi** [comment]
Addressed comments and added some more stuff while keeping those comments in mind

**Review by Preston4tw — APPROVED** [review]
LGTM

**Pull request merged**